tools/ci-build/runtime-versioner/src/tag.rs

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use crate::{command::CommandExt, repo::Repo};
+use anyhow::{anyhow, bail, Context, Result};
+use smithy_rs_tool_common::release_tag::ReleaseTag;
+use std::str::FromStr;
+use tracing::warn;
+
+/// Discovers and returns the tag of the previous release for the revision at HEAD.
+pub fn previous_release_tag(
+    repo: &Repo,
+    release_tags: &[ReleaseTag],
+    hint: Option<&str>,
```

Comment:
what's `hint` here? it seems like it's a manually specified tag?

Reply:
Yeah, it's a manually specified tag. Renamed to be clearer.